From 1f2577a54657699bebaee135e738478afc73a26f Mon Sep 17 00:00:00 2001
From: "hjnim.kim" <hjnim.kim@samsung.com>
Date: Wed, 2 Oct 2013 15:09:25 +0900
Subject: [PATCH] JQM: Popup resize event handler is called once more

If popup is received resize event during the orientationchange ( resize ) event, resize handler is called once more

Change-Id: Ib2d9d9ec282aee3a0ac02b929e60653c0d5c854c
---
 libs/js/jquery-mobile-1.2.0/js/widgets/popup.js |   12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/libs/js/jquery-mobile-1.2.0/js/widgets/popup.js b/libs/js/jquery-mobile-1.2.0/js/widgets/popup.js
index 3508291..5bc46d9 100644
--- a/libs/js/jquery-mobile-1.2.0/js/widgets/popup.js
+++ b/libs/js/jquery-mobile-1.2.0/js/widgets/popup.js
@@ -113,7 +113,7 @@ define( [ "jquery",
 		},
 
 		_resizeTimeout: function() {
-			if ( !this._maybeRefreshTimeout() && this.positionTo === "window" && this._isOpen ) {
+			if ( this.positionTo === "window" && this._isOpen ) {
 				// effectively rapid-open the popup while leaving the screen intact
 				this._trigger( "beforeposition" );
 				this._ui.container
@@ -135,6 +135,11 @@ define( [ "jquery",
 				}
 				this._maybeRefreshTimeout();
 			}
+
+			if ( !this._isOpen && this._isPreOpen && this.positionTo === "window" ) {
+					setTimeout( $.proxy( this, "_resizeTimeout" ), 200 );
+			}
+
 		},
 
 		_handleWindowOrientationchange: function( e ) {
@@ -198,6 +203,7 @@ define( [ "jquery",
 				_fallbackTransition: "",
 				_currentTransition: false,
 				_prereqs: null,
+				_isPreOpen: false,
 				_isOpen: false,
 				_tolerance: null,
 				_resizeData: null,
@@ -586,6 +592,7 @@ define( [ "jquery",
 			var self = this;
 
 			self._ui.container.addClass( "ui-popup-active" );
+			self._isPreOpen = false;
 			self._isOpen = true;
 			self._resizeScreen();
 
@@ -619,6 +626,9 @@ define( [ "jquery",
 					return false;
 				}());
 
+			// If popup is received resize event during the orientationchange ( resize ) event, resize handler is called once more
+			this._isPreOpen = true;
+
 			// Make sure options is defined
 			options = ( options || {} );
 
-- 
1.7.9.5

